src/main/java/org/apache/commons/io/output/DeferredFileOutputStream.java

```diff
@@ -306,6 +316,9 @@ public void close() throws IOException
      * @param out output stream to write to.
      * @throws IOException if this stream is not yet closed or an error occurs.
      */
+    // Checker issues false positive warning for using outputFile in FileInputStream object declaration
+    // isInMemory returns true when outputFile is null and buggy block is not executed. 
```

Comment:
This comment does not say why `memoryOutputStream` is non-null.  One of the two warnings is about a dereference of `memoryOutputStream`, so this comment does not help to understand why the code is guaranteed to be correct.

Reply:
This can be a bug, I will undo suppressing warning for `memoryOutputStream`